### PR TITLE
Bulk api error handling with new Json

### PR DESCRIPTION
**Pull request by msvinaykumar**
## Description

Following new format incorporated

```
{
  "status": "IN_PROGRESS",
  "total_experiments": 23,
  "processed_experiments": 22,
"experiments": [
    {
      "name": "prometheus-1|default|kube-system|coredns(deployment)|coredns",
      "notification": {},
      "recommendations": {
        "status": "unprocessed"
      }
    },
    {
      "name": "prometheus-1|default|kube-system|kindnet(deployment)|kindnet-cni",
      "notification": {},
      "recommendations": {
        "status": "processed"
      }
    },
    {
      "name": "prometheus-1|default|monitoring|kruize(deployment)|kruize",
      "notification": {},
      "recommendations": {
        "status": "processing"
      }
    },
    {
      "name": "prometheus-1|default|monitoring|kruize(deployment)|kruize",
      "recommendations": {
        "status": "failed",
        "notifications": {
          "400": {
            "type": "error",
            "message": "Not able to fetch metrics",
            "code": 400
          }
        }
      }
    },
    {
      "name": "prometheus-1|default|monitoring|kruize(deployment)|kruize",
      "recommendations": {
        "status": "failed",
        "notifications": {
          "400": {
            "type": "error",
            "message": "Not able to fetch metrics",
            "code": 400
          }
        }
      }
    }
  ] ,
    "job_id": "5798a2df-6c67-467b-a3c2-befe634a0e3a",
  "job_start_time": "2024-10-09T18:09:31.549Z",
  "job_end_time": null
}
 ```

Fixes # (issue)

### Type of change

- [x] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [x] New Test X
- [x] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: 

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [ ] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

### CODE REVIEW

**Comment by msvinaykumar** [comment]
Other examples

```

```

File: design/BulkAPI.md [hunk 1]
```diff
@@ -205,40 +200,37 @@ resource optimization in Kubernetes environments. Below is a breakdown of the JS
...
 
-    - **experiments**:
-        - **new**:
-            - **Type**: `Array of Strings`
-            - **Description**: List of new experiments that have been identified but not yet processed.
+  | Field                   | Type         | Description                                                              |
+      |-------------------------|--------------|--------------------------------------------------------------------------|
+  | `name`                  | `string`     | Name of the experiment, typically indicating a service name and deployment context. |
+  | `notification`          | `object`     | Notifications specific to this experiment (if any).                      |
+  | `recommendation`        | `object`     | Recommendation status and notifications specific to this experiment.     |
```
Comment (dinogun) [inline design/BulkAPI.md:213]: The table seems to be malformed

File: design/BulkAPI.md [hunk 2]
```diff
@@ -114,73 +114,68 @@ When verbose=true, additional detailed information about the job is provided.
...
+      "recommendation": {
+        "status": "processing",
+        "notification": {}
+      }
+    },
+    {
+      "name": "prometheus-1|default|monitoring|kruize(deployment)|kruize",
+      "recommendation": {
+        "status": "failed",
+        "notifications": {
```
Comment (dinogun) [inline design/BulkAPI.md:149]: I see `notifications` in some places and `notification` in some, lets keep it as `response` in all places

Comment (msvinaykumar) [reply design/BulkAPI.md:145]: I changed it to  `notifications` similar to updateRecommendations response .

**Review by dinogun — APPROVED** [review]
LGTM!

**Pull request merged**